droidlet/tools/hitl/turk_oncall/oncall_jobs.py

```diff
@@ -116,30 +148,181 @@ def run(self) -> None:
+        )
+        summary_stats.extend(
+            [
+                f"{key[0]}|{key[1]}|{COMMAND_LISTS[key[0]].replace('|', ',')}|{command_lut['feedback'][key]}|{command_lut['logs'][key]}\n"
+                for key in command_lut["result"].keys()
+                if command_lut["result"][key] == "no"
+            ]
+        )
+
+        # Log locally and upload results to S3
+        local_dir = f"{HITL_TMP_DIR}/{self._batch_id}/oncall_results"
+        os.makedirs(local_dir, exist_ok=True)
+
+        with open(os.path.join(local_dir, "summary_stats.txt"), "w") as f:
+            f.writelines(summary_stats)
+        with open(os.path.join(local_dir, "results.txt"), "w") as f:
+            f.writelines(results)
+        with open(os.path.join(local_dir, "softblock.txt"), "w") as f:
+            f.writelines(softblock)
+
+        logging.info(f"Uploading job results to S3 folder: {self._batch_id}")
+        s3.upload_file(
+            f"{local_dir}/summary_stats.txt",
+            S3_BUCKET_NAME,
+            f"{self._batch_id}/summary_stats.txt",
+        )
+        s3.upload_file(
+            f"{local_dir}/results.txt",
+            S3_BUCKET_NAME,
+            f"{self._batch_id}/results.txt",
+        )
+        s3.upload_file(
+            f"{local_dir}/softblock.txt",
+            S3_BUCKET_NAME,
+            f"{self._batch_id}/softblock.txt",
+        )
+
+    def process_s3_logs(self) -> dict:
+        """
+        This reads the local logs directory after all tarfiles have been extracted.
+        It returns a map of Mephisto Agent IDs to the directory names of the logs they produced.
+        """
+        s3_logs_dir = os.path.join(HITL_TMP_DIR, f"{self._batch_id}/turk_logs")
+        parsed_logs_dir = os.path.join(HITL_TMP_DIR, f"{self._batch_id}/parsed_turk_logs")
+        os.makedirs(s3_logs_dir, exist_ok=True)
+        os.makedirs(parsed_logs_dir, exist_ok=True)
+
+        subprocess.call(
+            [f"aws s3 sync {S3_ROOT}/{self._batch_id}/interaction {s3_logs_dir}"],
+            shell=True,
+        )
+        logging.info("Waiting 2min for S3 directory sync to finish")
+        time.sleep(120)
+
+        # Extract tarfiles
+        read_s3_bucket(s3_logs_dir, parsed_logs_dir)
+
+        # Build a map of agent ids to logs directories
+        agent_logs_map = {}
+        for json_path in glob.glob(f"{parsed_logs_dir}/**/job_metadata.json"):
+            with open(json_path, "r") as f:
+                js = json.load(f)
+
+            dir = os.path.basename(os.path.dirname(json_path))
+            agent_logs_map[js["mephisto_agent_id"]] = dir
+
+        return agent_logs_map
+
 
 if __name__ == "__main__":
     runner = TaskRunner()
-    ocj = OnCallJob(instance_num=2, image_tag="oncall_v1", task_name="oncall_t2", timeout=30)
+    ocj = OnCallJob(
+        instance_num=2,
+        image_tag="oncall_v1",
+        task_name="oncall_t2",
```

Comment:
I wonder if the line 323 `task_name="oncall_t2"` should be deleted here - since it has been removed from the init function for OnCallJob class.

Reply:
good catch, thanks!